includes/data-stores/class-wc-admin-reports-orders-stats-data-store.php

```diff
@@ -515,24 +515,56 @@ protected static function get_num_items_sold( $order ) {
 	 * @return bool
 	 */
 	protected static function is_returning_customer( $order ) {
-		global $wpdb;
-		$customer_id        = WC_Admin_Reports_Customers_Data_Store::get_customer_id_by_user_id( $order->get_user_id() );
-		$orders_stats_table = $wpdb->prefix . self::TABLE_NAME;
+		$customer_id = WC_Admin_Reports_Customers_Data_Store::get_customer_id_by_user_id( $order->get_user_id() );
 
 		if ( ! $customer_id ) {
 			return false;
 		}
 
-		$customer_orders = $wpdb->get_var(
+		$oldest_orders = WC_Admin_Reports_Customers_Data_Store::get_oldest_orders( $customer_id );
+
+		if ( empty( $oldest_orders ) ) {
+			return false;
+		}
+
+		$first_order = $oldest_orders[0];
+
+		// Order is older than previous first order.
+		if ( $order->get_date_created() < new WC_DateTime( $first_order->date_created ) ) {
+			self::set_customer_first_order( $customer_id, $order->get_id() );
+			return false;
+		}
+		// First order date has changed and next oldest is now the first order.
```

Comment:
To do this logic here, should we first verify that `$oldest_orders[1]` exists, and if so maybe assign it to a variable to keep things more legible here? `$second_order` maybe? I see the check on 541, but maybe we can skip the entire logic block by checking that earlier.

Reply:
Seems like a good idea.  Implemented in https://github.com/woocommerce/wc-admin/pull/1517/commits/0994f3bbf7e67d058b39bb91d2559206f85a45c0